pkgs/build-support/build-typst-package.nix

```
@@ -0,0 +1,54 @@
+{
+  lib,
+  stdenvNoCC,
+}:
+
+/**
+  `buildTypstPackage` is a help builder for typst packages.
+
+  # Inputs
+
+    `attrs`
+    : attrs for stdenvNoCC.mkDerivation + typstDeps (a list of `buildTypstPackage` derivations)
+
+  # Example
+  ```nix
+  { buildTypstPackage, typstPackages }:
+
+  buildTypstPackage {
+    pname = "example";
+    version = "0.0.1";
+    src = ./.;
+    typstDeps = with typstPackages; [ oxifmt ];
+  }
+  ```
+*/
+
+{
+  typstDeps ? [ ],
+  ...
+}@attrs:
+let
+  cleanAttrs = lib.flip lib.removeAttrs [ "typstDeps" ];
+in
+stdenvNoCC.mkDerivation (
+  (cleanAttrs attrs)
+  // {
+    name = "typst-package-${attrs.pname}-${attrs.version}";
+
+    installPhase =
+      let
+        outDir = "$out/lib/${attrs.pname}/${attrs.version}";
```

If Typst adopts some foreign-function interface and other projects come with their "Typst bindings," a Typst package might be packaged as part of other libraries, and it would be great to place them in different sub-directories.
```suggestion
        outDir = "$out/lib/typst_packages/${attrs.pname}/${attrs.version}";
```

---

Sure, that makes sense. Fixed